### PR TITLE
Fork artifacts into test vs production (wip)

### PR DESCRIPTION
**Pull request by sjmiles**
## Primary Goal

Work towards simplifying and regularizing how Particles are written and to try to maximize the leverage of auto-composition.

### Some Practical Targets

1. Modernize demo particles (in particular `Restaurants` code is really old)
2. Minimize manifest boilerplate
3. Simplifications to reduce maintenance

### Things Happened

There are runtime tests that depend on artifacts (manifests, particles) to function. Rebuilding demo artifacts caused significant incidental test breakage. I suggest we need better differentiation between things under test and test dependencies, e.g. clarity between unit tests and integration tests.

Towards this end, I forked Restaurants and Products folders (in `artifacts/`) so that the legacy code used by the unit tests are in TestRestaraunts and TestProducts (the integration [Selenium] tests still use the primary artifacts). Probably we should move Test* into the runtime/test folder, and/or simplify those contents.

### Multiplexer and Lists

I've been exploring various ways to simplify utilization of Multiplexer, this PR contains the latest volley. To reduce cognitive load, I moved the List code out of `artifacts/Common` into `artifacts/List` (this move and various renames also impacted the runtime tests). All of this naming is subject to bikeshedding.

Latest thinking: each particular List shape has it's own `.recipes` file which provides recipes for generating that List with it's particular Multiplexer infrastructure. In this way, Multiplexer complexity is encapsulated. See https://github.com/sjmiles/arcs/blob/fork-artifacts/artifacts/List/Items.recipes for an example. 

Note: I'm being coy about the notion of a List _shape_: this is partly rendered geometry, partly data requirements, and some other stuff. For example, `Annotation.recipes` defines a multiplexer for working with 'Annotation' set-slots, but doesn't actually express a new type of list.

### Other Moving

The modern Detail code  was moved out of Common and into Layout.

### CODE REVIEW

File: artifacts/Restaurants/source/ReservationForm.js [hunk 1]
```diff
@@ -247,17 +240,17 @@ ${styles}
...
     }
-    _onPartySizeChanged(e, state) {
+    onPartySizeChanged(e, state) {
       let newEvent = Object.assign({}, state.currentEvent || {});
       newEvent.participants = Number(e.data.value);
-      this._storeNewEvent(newEvent);
+      this.storeNewEvent(newEvent);
     }
-    _storeNewEvent(newEvent) {
+    storeNewEvent(newEvent) {
```
Comment (mmandlis) [inline artifacts/Restaurants/source/ReservationForm.js]: need to update line 153 in this file from `_storeNewEvent` to `storeNewEvent`

**Comment by mmandlis** [comment]
Did you intend for the demo-flow-test to use the real Common & List particles? This is what TestProducts particles are using. Maybe it is best to just go ahead and move the test particles under runtime/test or similar and completely disconnect it from the real ones (under artifacts/) to avoid confusion?

**Comment by sjmiles** [comment]
re @mmandlis 
> Did you intend for the demo-flow-test to use the real Common & List particles?

Nope. I didn't do a comprehensive job at all, since I figured we had to discuss these notions. E.g., as we discussed, having Test* under Artifacts isn't really intended either.

**Comment by sjmiles** [comment]
@mmandlis, re: demos

As we discussed, if we veer towards intelligent composition of recipes (away from hand-built or learned recipes) it creates different stresses on the system.

Specifically, to recreate the existing Demo flows with minimal recipes will require extra work. I don't believe it's possible to have both right away.

That &nearbyRestaurants affects search wasn't clear to me, we should probably discuss. Fwiw, it seems a bit strange to mix natural-language and machine-readable text into one subsystem.

**Comment by sjmiles** [comment]
No longer relevant.

**Pull request closed without merging**